tensorboard/compat/tensorflow_stub/io/gfile.py

```diff
@@ -395,8 +483,37 @@ def __next__(self):
     def next(self):
         return self.__next__()
 
+    def flush(self):
+        if self.write_temp is not None:
+            # read temp file from the beginning
+            self.write_temp.flush()
+            self.write_temp.seek(0)
+            chunk = self.write_temp.read()
+            if chunk is not None:
+                # use append if filesystem supports it
+                if self.fs_supports_append:
+                    if not self.write_started:
```

Comment:
I don't think you actually need to track a `write_started`, do you?  You can call append() even if the file doesn't already exist and in that case it will create it.

Reply:
Actually, we do need this. If a file already has content (thank you for the test coverage suggestions) then we will just append to that contents on first write rather than clearing it out. This is the most efficient way I could see to handle this. Let me know your thoughts, though.

Reply:
Also, this code will move up into `write()`.